clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -2064,6 +2064,9 @@ llvm::Value *CodeGenFunction::EmitToMemory(llvm::Value *Value, QualType Ty) {
 
   if (Ty->isExtVectorBoolType()) {
     llvm::Type *StoreTy = convertTypeForLoadStore(Ty, Value->getType());
+    if (getLangOpts().HLSL)
```

Comment:
Do we need to check HLSL here? We should be able to instead check that StoreTy refers to a smaller integer type than Value->getType() right?

Reply:
Yes, I guess I'd be worried that would cause unintended consequences for other languages. Maybe that is unlikely.

Reply:
Does this point apply to the 'isPackedBoolVectorType' function?